osquery/tables/system/linux/system_info.cpp

```diff
@@ -40,15 +41,12 @@ QueryData genSystemInfo(QueryContext& context) {
     }
   }
 
-  // Can parse /proc/cpuinfo or /proc/meminfo for this data.
-  static long cores = sysconf(_SC_NPROCESSORS_CONF);
-  if (cores > 0) {
-    r["cpu_logical_cores"] = INTEGER(cores);
-    r["cpu_physical_cores"] = INTEGER(cores);
-  } else {
-    r["cpu_logical_cores"] = "-1";
-    r["cpu_physical_cores"] = "-1";
-  }
+  auto logical_cores = boost::thread::hardware_concurrency();
+  r["cpu_logical_cores"] = (logical_cores > 0) ? INTEGER(logical_cores) : "-1";
+
+  auto physical_cores = boost::thread::physical_concurrency();
```

Comment:
**static**

Reply:
Number of physical cores can actually change during runtime. CPU hotplug is a thing.

Reply:
Good catch.  Completely overlooked this.